source/MaterialXCore/Element.cpp

```diff
@@ -104,6 +104,24 @@ string Element::getNamePath(ConstElementPtr relativeTo) const
     return res;
 }
 
+ElementPtr Element::getDescendant(const string& path)
+{
+    const vector<string> elementNames = splitString(path, NAME_PATH_SEPARATOR);
+    ElementPtr currentElement = getSelf();
+    if (elementNames.empty())
```

Comment:
Just a minor note, but you could actually omit this entire early out (including the "if" test"), since the range-based loop below will be skipped if the vector is empty.

Reply:
Good point.